Library/Formula/h2o.rb

```diff
@@ -0,0 +1,82 @@
+class H2o < Formula
+  homepage "https://github.com/h2o/h2o/"
+  url "https://github.com/h2o/h2o/archive/v0.9.0.tar.gz"
+  sha1 "37b84750900cdb56c3be574477d6f3327d92a4d7"
+  head "https://github.com/h2o/h2o.git"
+
+  option "with-libuv", "Build the H2O library as well as the executable."
+
+  depends_on "cmake" => :build
+  depends_on "libyaml"
+  depends_on "openssl"
+  depends_on "libuv" => :optional
+
+  def install
+    system "cmake", ".", *std_cmake_args
+
+    if build.with? "libuv"
+      system "make", "libh2o"
+      lib.install "libh2o.a"
+    end
+
+    system "make", "install"
+
+    mkdir_p etc/"h2o"
+    mkdir_p var/"h2o"
+    (etc+"h2o").install "examples/h2o/alternate.crt", "examples/h2o/alternate.key"
+    (etc+"h2o").install "examples/h2o/server.crt", "examples/h2o/server.key"
+    (var+"h2o").install "examples/doc_root/index.html"
+    (etc+"h2o/h2o.conf").write conf_example
+  end
+
+  def conf_example; <<-EOS.undent
+    listen: 8080
+    listen:
+       port: 8081
+       ssl:
+         certificate-file: #{etc}/h2o/server.crt
+         key-file: #{etc}/h2o/server.key
+    hosts:
+      "127.0.0.1.xip.io:8080":
+        paths:
+          /:
+            file.dir: #{var}/h2o/
+    EOS
+  end
+
+  def caveats; <<-EOS.undent
+    Example certificates, keys and configuration files have been placed
+    in #{etc}/h2o. These example keys and certificates are available
+    publicly online and *should not* be trusted as secure. Use your own
+    generated keys and certs for actual production use, please.
+    EOS
+  end
+
+  plist_options :manual => "h2o"
+
+  def plist; <<-EOS.undent
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+      <dict>
+        <key>Label</key>
+        <string>#{plist_name}</string>
+        <key>RunAtLoad</key>
+        <true/>
+        <key>KeepAlive</key>
+        <true/>
+        <key>ProgramArguments</key>
+        <array>
+            <string>#{opt_bin}/h2o</string>
+            <string>-c</string>
+            <string>#{etc}/h2o/h2o.conf</string>
+        </array>
+      </dict>
+    </plist>
+    EOS
+  end
+
+  test do
+    system bin/"h2o", "--version"
```

Comment:
To catch this before Mike does, I'm afraid this test can't be made any better without actually starting the HTTP server :wink: